RCNN_tf.py

```diff
@@ -116,209 +176,292 @@ def loop_body(x, recurrent_states, output):
 
-        # dropout probability placeholder
+        # Dropout probability placeholder
         self.rate_placeholder = tf.placeholder(PRECISION_TF)
-        # number of data placeholder
+
+        # Number of data placeholder
         self.num_data_placeholder = tf.placeholder(tf.int64)
 
         # Create data set objects
         training_data_set = tf.data.Dataset.from_tensor_slices((
             self.input_placeholder,
-            self.output_placeholder
-        )).shuffle(buffer_size=shuffle_buffer_size).repeat().batch(batch_size=self.num_data_placeholder)
+            self.output_placeholder)
+            ).shuffle(buffer_size=shuf_buf_size
+            ).repeat(
+            ).batch(batch_size=self.num_data_placeholder)
+
         test_data_set = tf.data.Dataset.from_tensor_slices((
             self.input_placeholder,
-            self.output_placeholder,
-        )).batch(batch_size=self.num_data_placeholder)
+            self.output_placeholder,)
+            ).batch(batch_size=self.num_data_placeholder)
 
         # Create Iterator
-        data_iterator = tf.data.Iterator.from_structure(training_data_set.output_types,
-                                                         training_data_set.output_shapes)
+        data_iterator = tf.data.Iterator.from_structure(
+            training_data_set.output_types,
+            training_data_set.output_shapes
+        )
+
         # Initialize iterators and get input and output placeholder variables
         self.train_init_op = data_iterator.make_initializer(training_data_set)
         self.test_init_op = data_iterator.make_initializer(test_data_set)
+
         # Defines the pipeline and creates a pointer to the next data point
         features, labels = data_iterator.get_next()
 
-        with tf.device(self.device_name):
-            # Net definition
+        ### Definition of the neural network
+        with tf.device(self.processing_unit):
             # First convolutional layer
             conv_layer = convolutional_layer(
-                features,
-                num_input_channels=input_shape[-1],
-                filter_shape=conv_layer_filter_shape,
+                input_data=features,
+                filter_shape=conv_filter_shape,
+                num_input_chans=input_shape[-1],
                 num_filter=num_filter,
                 name='conv_layer_1'
             )
 
+            # First pooling layer
             pooling_1 = pooling_layer(
                 input_data=conv_layer,
-                pool_shape=pooling_shape,
-                stride=pooling_stride_shape
+                pool_shape=pool_shape,
+                stride=pool_stride_shape
             )
 
-            # Recurrent convolutional layers
+            # First recurrent convolutional layer
             rcl_layer_1 = rcl(
                 input_data=pooling_1,
-                num_input_channels=num_filter,
+                num_input_chans=num_filter,
```

Comment:
Fine for now, but in future I would like to avoid renaming variables, if there is no reason for that. It is totally appropriate when they are deprecated, misleading, or not descriptive.
This, however, adds points of failure and blows up the changes on github which are needed to be reviewed.